testsuite/testsuite_impl/my_test/my_test_.go

```diff
@@ -0,0 +1,202 @@
+	"github.com/kurtosis-tech/kurtosis-client/golang/lib/services"
+	"github.com/kurtosis-tech/kurtosis-testsuite-api-lib/golang/lib/testsuite"
+	"github.com/palantir/stacktrace"
+	"github.com/sirupsen/logrus"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+const (
+	ethereumGoClientImageName = "ethereum/client-go"
+	ethereumGoClientPort      = 8565
+	rpcPort                   = 8545
+	adminInfoRpcCall          = `{"jsonrpc":"2.0","method": "admin_nodeInfo","params":[],"id":67}`
+
+	waitForStartupTimeBetweenPolls = 1 * time.Second
+	waitForStartupMaxPolls         = 90
+
+	serviceID = "my-eth-client"
+)
+
+type MyTest struct{}
+
+type NodeInfoResponse struct {
+	Result NodeInfo `json:"result"`
+}
+
+type NodeInfo struct {
+	Enode string `json:"enode"`
+}
+
+func (test MyTest) Configure(builder *testsuite.TestConfigurationBuilder) {
+	builder.WithSetupTimeoutSeconds(360).WithRunTimeoutSeconds(360)
+}
+
+func (test MyTest) Setup(networkCtx *networks.NetworkContext) (networks.Network, error) {
+	containerCreationConfig, runConfigFunc := getMyServiceConfigurations()
+
+	serviceContext, hostPortBindings, err := networkCtx.AddService(serviceID, containerCreationConfig, runConfigFunc)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred adding the Ethereum Go Client service")
+	}
+
+	firstNodeUp := false
+	for pollCount := 0; pollCount < waitForStartupMaxPolls; pollCount++ {
+		enodeAddress, err := getEnodeAddress(serviceContext.GetIPAddress())
+		if err == nil {
+			firstNodeUp = true
+			logrus.Infof("Enode address: %v", enodeAddress)
+			break
+		}
+		time.Sleep(waitForStartupTimeBetweenPolls)
+	}
+	if !firstNodeUp {
+		return nil, stacktrace.Propagate(err, "First geth node failed to come up")
+	}
+
+	logrus.Infof("Added Ethereum Go Client service with host port bindings: %+v", hostPortBindings)
+	return networkCtx, nil
+}
+
+func (test MyTest) Run(uncastedNetwork networks.Network) error {
+	// Necessary because Go doesn't have generics
+	castedNetwork := uncastedNetwork.(*networks.NetworkContext)
+
+	serviceCtx, err := castedNetwork.GetServiceContext(serviceID)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred getting the Ethereum Go Client service context")
+	}
+	logrus.Infof("Got service context for Ethereum Go Client service '%v'", serviceCtx.GetServiceID())
+
+	gethClient, err := getClient(serviceCtx.GetIPAddress())
+	if err != nil {
+		return stacktrace.Propagate(err, "Failed to get a gethClient from first node.")
+	}
+	defer gethClient.Close()
+
+	networkId, err := gethClient.NetworkID(context.Background())
+	if err != nil {
+		return stacktrace.Propagate(err, "Failed to get network ID")
+	}
+	logrus.Infof("Chain ID: %v", networkId)
+
+	exitCode, logOutput, err := serviceCtx.ExecCommand([]string{"/bin/sh", "-c",
+		fmt.Sprintf("printf \"passphrase\\npassphrase\\n\" | geth attach /tmp/geth.ipc --exec 'personal.newAccount()'"),
+	})
+	if err != nil || exitCode != 0 {
```

Comment:
Noice - I'd recommend splitting these into two `if` statements (`err != nil` first), so that the error you returned can be more specific. E.g. if `err != nil`, then your `logOutput` is likely going to be empty